com.ibm.wala.core/src/com/ibm/wala/dataflow/IFDS/LocalSummaryEdges.java

```diff
@@ -27,34 +26,31 @@
 public class LocalSummaryEdges {
 
   /**
-   * A map from integer n -> (IBinaryNonNegativeIntRelation)
+   * This code was modified by Lucja Kot in December 2016 after identifying it
+   * as a heavy hotspot during profiling.
    * 
-   * Let s_p be an entry to this procedure, and x be an exit. n is a integer which uniquely identifies an (s_p,x) relation. For any
-   * such n, summaries[n] gives a relation R=(d1,d2) s.t. (<s_p, d1> -> <x,d2>) is a summary edge.
+   * The old code used two maps, one taking (s_p, x) to an integer n and one
+   * taking n to an IBinaryNonNegativeIntRelation. In the refactoring, both were
+   * merged into a single java HashMap.
    * 
-   * Note that this representation is a little different from the representation described in the PoPL 95 paper. We cache summary
-   * edges at the CALLEE, not at the CALLER!!! This allows us to avoid eagerly installing summary edges at all call sites to a
-   * procedure, which may be a win.
+   * Let s_p be an entry to this procedure, and x be an exit. Let (s_p,x) be an
+   * entry-exit pair, and let l := the long whose high word is s_p and low word
+   * is x. summaryMap.getl(l) gives a relation R=(d1,d2) s.t. (<s_p, d1> ->
+   * <x,d2>) is a summary edge.
    * 
-   * we don't technically need this class, since this information is redundantly stored in LocalPathEdges. However, we're keeping it
-   * cached for now for more efficient access when looking up summary edges.
+   * Note that this representation is a little different from the representation
+   * described in the PoPL 95 paper. We cache summary edges at the CALLEE, not
+   * at the CALLER!!! This allows us to avoid eagerly installing summary edges
+   * at all call sites to a procedure, which may be a win.
+   * 
+   * we don't technically need this class, since this information is redundantly
+   * stored in LocalPathEdges. However, we're keeping it cached for now for more
+   * efficient access when looking up summary edges.
    * 
    * TODO: more representation optimization.
    */
-  private final SparseVector<IBinaryNaturalRelation> summaries = new SparseVector<IBinaryNaturalRelation>(1, 1.1f);
-
-  /**
-   * Let (s_p,x) be an entry-exit pair, and let l := the long whose high word is s_p and low word is x.
-   * 
-   * Then entryExitMap(l) is an int which uniquely identifies (s_p,x)
-   * 
-   * we populate this map on demand!
-   */
-  private final static int UNASSIGNED = -1;
-
-  private final SparseLongIntVector entryExitMap = new SparseLongIntVector(UNASSIGNED);
-
-  private int nextEntryExitIndex = 0;
+  
+  private final HashMap<Long, IBinaryNaturalRelation> summaryMap = new HashMap<Long,IBinaryNaturalRelation>();
```

Comment:
summaries and summaryMap represent the same information.  The name change is just for clarity, I assume?  That's fine.  Just double-checking that I'm not missing something subtle.